### DISCUSSION TITLE
[Store Persist] How to create a new store with a dynamic naming mechanism without merging from the old state

### DISCUSSION THREAD

**Discussion by hassanedelbi** [pmndrs/zustand#2925] [Q&A]
From the docs: 
[setOptions](https://zustand.docs.pmnd.rs/integrations/persisting-store-data#setoptions)
Type: (newOptions: Partial<PersistOptions>) => void

Changes the middleware options. Note that the new options will be merged with the current ones.
i am using persist like this 

`
useEffect(() => {

	useUserPreferencesStore.persist.setOptions({
		name: user
			? `user-${user.id}preferences-storage`
			: "user-preferences-storage-anonymous",
	});
}, [user]);
`

I am saving some preferences per user in a separate localStorage key 

**Expected**:
for example when the user1 logs in i have user-1-preferences-storage
then when the user logs in with user 2 it s user-2-preferences-storage

**Current Behavior**

currently when the user logs in with user 1 i ll have  user-1-preferences-storage
when i switch to user 2 and then doing .persist.setOptions it s copying the user-1 data to user-2 and reseting user-1 preferences to initial.

**Comment by dbritto-dev** [comment]
@hassanedelbi would you mind creating a minimal repro on stackblitz? I'd be happy to help
